File: uStock-Development-Android/taskThree_models/Data Structures/CompetitionModels/Competition.swift
//
//  Competition.swift
//  uStockIOS-V2-AI
//
//  Created by Gal Cohavy on 5/15/23.
//

import Foundation

public struct Competition: Codable, Identifiable {
    public var id: String
    let name: String
    let description: String?
    let goalType: String
    let competitionGoal: String?
    let timeConstraint: Int?
    let endDate: Date?
    let participantsModel: String
    let participants: [String]
    
    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case name, description, goalType, competitionGoal, timeConstraint, endDate, participantsModel, participants
    }
}


public struct CompetitionProgressUpdate {
    let competition: String
    let team: String
    let post: String? //in case this is an update with no post
    let progress: Int? //in case this is an update with no progress
    let date: Date
}




public struct CompetitionResponse: Codable { //for matching to node api response which gives {competition: Competition object response}.
    let competition: Competition
}

File: uStock-Development-Android/app/src/main/java/data_structures/competition_model/Competition.kt
package data_structures.competition_model
import data_structures.DateSerializer
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import java.util.Date

@Serializable
data class Competition(
    @SerialName("_id") val id: String,
    val name: String,
    val description: String?,
    @SerialName("goalType") val goalType: String,
    @SerialName("competitionGoal") val competitionGoal: String?,
    @SerialName("timeConstraint") val timeConstraint: Int?,
    @Serializable(with = DateSerializer::class)
    @SerialName("endDate") val endDate: Date?,
    @SerialName("participantsModel") val participantsModel: String,
    val participants: List<String>
)

@Serializable
data class CompetitionResponse(
    val competition: Competition
)
